lib/base-compiler.ts

```diff
@@ -775,7 +775,16 @@ export class BaseCompiler {
         );
     }
 
-    orderArguments(options, inputFilename, libIncludes, libOptions, libPaths, libLinks, userOptions, staticLibLinks) {
+    orderArguments(
+        options: string[],
+        inputFilename: string,
+        libIncludes: any[], // TODO: Fix this type
```

Comment:
I think these are all `string[]`, considering they have to fit into `options.concat()`, right?

Reply:
right, hmm..

Reply:
Will check on Monday

Reply:
So, it all gets compacted in the end so we can `flatMap` those functions without problems.
BUT, in order to do so, we need to fix `libraries.interfaces.ts` because both `path`  and `options` are `string[]` and not `string` (See for example https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/options-handler.js#L198)

Fixing that in this PR will make it too confusing so I don't know if we should skip it, merge this PR as is, and create another PR to properly handle this issues

Reply:
:+1: